api/src/main/java/com/epam/pipeline/manager/cloud/CloudInstancePriceService.java

```diff
@@ -36,4 +37,17 @@
     List<InstanceOffer> refreshPriceListForRegion(T region);
     double getSpotPrice(String instanceType, T region);
     double getPriceForDisk(List<InstanceOffer> offers, int instanceDisk, String instanceType, T region);
+    List<InstanceType> getAllInstanceTypes(Long id, Boolean spot);
```

Comment:
How does this method selects instance types by provider, if `regionId` is not specified?
Please, call argument `regionId`
If `spot` argument is required, make it `boolean` to prevent NPE